### ISSUE TITLE
_nodes/http: http_address missing in 5.0.0-alpha5

### ISSUE THREAD

**Issue by olivere** [elastic/elasticsearch#20091]
**Elasticsearch version**: 5.0.0-alpha5

**Plugins installed**: []

**JVM version**: 1.8.0_66

**OS version**: Darwin 15.6.0

**Description of the problem including expected versus actual behavior**:

As the author of a Go client, I rely on sniffing nodes in the cluster. This is currently being done via `_nodes/http` and picking the `http_address` (or `https_address`). However, these fields are missing in 5.0.0-alpha5 here. The contents of these fields have been changed from time to time. But they've never been missing entirely. I also cannot find a reference in the docs. Is this a regression or will the method for resolving the HTTP(S) endpoint change in 5.0?

**Steps to reproduce**:
```sh
$ curl -s 'localhost:9200/_nodes/http?pretty'
{
  "_nodes" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "cluster_name" : "elasticsearch",
  "nodes" : {
    "M8N4pvE2QUmwsci_RdK4cg" : {
      "name" : "M8N4pvE",
      "transport_address" : "127.0.0.1:9300",
      "host" : "127.0.0.1",
      "ip" : "127.0.0.1",
      "version" : "5.0.0-alpha5",
      "build_hash" : "d327dd4",
      "roles" : [
        "master",
        "data",
        "ingest"
      ],
      "http" : {
        "bound_address" : [
          "[::1]:9200",
          "127.0.0.1:9200",
          "172.17.0.2:9200"
        ],
        "publish_address" : "127.0.0.1:9200",
        "max_content_length_in_bytes" : 104857600
      }
    }
  }
}
```

**Comment by rjernst** [comment]
This was removed in #19218. However, the http address is still available under the nodes info API. https://www.elastic.co/guide/en/elasticsearch/reference/2.3/cluster-nodes-info.html

**Comment by rjernst** [comment]
Let me rephrase a bit: if you look at the reponse you pasted above, see the "http" section, and "publish_address"
